src/FilamentDevelopersLogin.php

```diff
@@ -21,26 +21,28 @@ public function login(Panel $panel, FilamentDeveloperLoginsPlugin $plugin, strin
             $panel->auth()->logout();
         }
 
+        /** @var ?\Illuminate\Contracts\Auth\Authenticatable $model */
         $model = (new ($plugin->getModelClass()))
             ->where($plugin->getColumn(), $credentials)
-            ->firstOrFail();
-
-        $panel->auth()->login($model);
-
-        if (
-            ($model instanceof FilamentUser) &&
-            (! $model->canAccessPanel($panel))
-        ) {
-            $panel->auth()->logout();
+            ->first();
 
+        if (! $model) {
             throw ValidationException::withMessages([
-                'developer-logins-failed' => __('filament-developer-logins::auth.messages.failed'),
+                'developer-logins-failed' => __('filament-developer-logins::auth.messages.failed_not_found'),
             ]);
         }
 
-        session()->regenerate();
+        if (! $model instanceof FilamentUser || $model->canAccessPanel($panel)) {
+            $panel->auth()->login($model);
```

Comment:
I reordered some logic here: login no longer takes place if it shouldnt have happened at all.

Previously, a `User implements FilamentUser` with a `canAccessPanel() === false` would still execute the `$panel->auth()->login($model)` after which it would be immediately logged out.

Reply:
The order of events was taken from Filament. For example, you need to login a user to check if it has permissions to view the panel.

Reply:
fixed